test/test-config.sh

```diff
@@ -163,8 +163,8 @@ fi
 # < 1.19) and because there were random connection failures to node ports during sanity
 # testing (Kubernetes 1.19.0)
 : ${TEST_FEATURE_GATES:=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true\
-$(case ${TEST_KUBERNETES_VERSION} in 1.1[0-5]) ;; *) echo ',EndpointSlice=false';; esac)\
-$(case ${TEST_KUBERNETES_VERSION} in 1.19) echo ',CSIStorageCapacity=true,GenericEphemeralVolume=true,EndpointSliceProxying=false';; esac)\
+$(case ${TEST_KUBERNETES_VERSION} in 1.1[0-5]) ;; 1.20) ;; *) echo ',EndpointSlice=false,EndpointSliceProxying=false';; esac)\
+$(case ${TEST_KUBERNETES_VERSION} in 1.19 | 1.20) echo ',CSIStorageCapacity=true,GenericEphemeralVolume=true';; esac)\
```

Comment:
There's still no explanation why 1.20 needs to be configured this way. What's wrong with keeping the endpoint feature disabled?

The first line can be shortened to:
```
$(case ${TEST_KUBERNETES_VERSION} in 1.1[6-9]) echo ',EndpointSlice=false,EndpointSliceProxying=false';; esac)\
```

Reply:
> There's still no explanation why 1.20 needs to be configured this way. What's wrong with keeping the endpoint feature disabled?

I didn't investigate much, but my observation was, `kube-proxy` was failing to connect to cluster api with some endpoint errors, that was when I tried with reenabling them. And as the issue mentioned in the comment was fixed, so I didn't try much investigating this.

Reply:
To my surprise, the same error message is showing even in v19.1 cluster with the current change, this is what observed while working with v1.20:

```
$ sudo grep feature-gates /etc/kubernetes/manifests/kube-apiserver.yaml 
    - --feature-gates=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true,EndpointSlice=false,EndpointSliceProxying=false,CSIStorageCapacity=true,GenericEphemeralVolume=true
   
$ kubectl logs -n kube-system kube-proxy-kwj6m
I0122 09:20:52.758697       1 node.go:136] Successfully retrieved node IP: 172.17.0.4
I0122 09:20:52.758844       1 server_others.go:111] kube-proxy node IP is an IPv4 address (172.17.0.4), assume IPv4 operation
W0122 09:20:55.569023       1 server_others.go:579] Unknown proxy mode "", assuming iptables proxy
I0122 09:20:55.569301       1 server_others.go:186] Using iptables Proxier.
W0122 09:20:55.569328       1 server_others.go:456] detect-local-mode set to ClusterCIDR, but no cluster CIDR defined
I0122 09:20:55.569335       1 server_others.go:467] detect-local-mode: ClusterCIDR , defaulting to no-op detect-local
I0122 09:20:55.570015       1 server.go:650] Version: v1.19.1
I0122 09:20:55.570906       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I0122 09:20:55.570957       1 conntrack.go:52] Setting nf_conntrack_max to 131072
I0122 09:20:55.572328       1 conntrack.go:83] Setting conntrack hashsize to 32768
I0122 09:20:55.579243       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I0122 09:20:55.579525       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I0122 09:20:55.579958       1 config.go:315] Starting service config controller
I0122 09:20:55.580084       1 shared_informer.go:240] Waiting for caches to sync for service config
I0122 09:20:55.580343       1 config.go:224] Starting endpoint slice config controller
I0122 09:20:55.580469       1 shared_informer.go:240] Waiting for caches to sync for endpoint slice config
E0122 09:20:55.585394       1 reflector.go:127] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1beta1.EndpointSlice: failed to list *v1beta1.EndpointSlice: endpointslices.discovery.k8s.io is forbidden: User "system:serviceaccount:kube-system:kube-proxy" cannot list resource "endpointslices" in API group "discovery.k8s.io" at the cluster scope
I0122 09:20:55.680448       1 shared_informer.go:247] Caches are synced for service config 
E0122 09:20:56.984422       1 reflector.go:127] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1beta1.EndpointSlice: failed to list *v1beta1.EndpointSlice: endpointslices.discovery.k8s.io is forbidden: User "system:serviceaccount:kube-system:kube-proxy" cannot list resource "endpointslices" in API group "discovery.k8s.io" at the cluster scope
....
```
And coredns and calico is not able to start:
```
$ kubectl  get po -o wide -n kube-system
NAME                                                READY   STATUS              RESTARTS   AGE   IP           NODE                         NOMINATED NODE   READINESS GATES
calico-kube-controllers-744cfdf676-7xxht            0/1     ContainerCreating   0          25m   <none>       pmem-csi-pmem-govm-worker2   <none>           <none>
calico-node-7ftbn                                   0/1     CrashLoopBackOff    7          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
calico-node-bh77f                                   0/1     CrashLoopBackOff    7          21m   172.17.0.4   pmem-csi-pmem-govm-worker2   <none>           <none>
calico-node-qwlx6                                   0/1     Running             9          21m   172.17.0.3   pmem-csi-pmem-govm-worker1   <none>           <none>
calico-node-szq6b                                   0/1     Running             9          21m   172.17.0.5   pmem-csi-pmem-govm-worker3   <none>           <none>
coredns-f9fd979d6-4xcjm                             0/1     ContainerCreating   0          25m   <none>       pmem-csi-pmem-govm-worker1   <none>           <none>
coredns-f9fd979d6-m24q6                             0/1     ContainerCreating   0          25m   <none>       pmem-csi-pmem-govm-worker3   <none>           <none>
etcd-pmem-csi-pmem-govm-master                      1/1     Running             0          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
kube-apiserver-pmem-csi-pmem-govm-master            1/1     Running             0          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
kube-controller-manager-pmem-csi-pmem-govm-master   1/1     Running             2          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
kube-proxy-65vzf                                    1/1     Running             0          21m   172.17.0.3   pmem-csi-pmem-govm-worker1   <none>           <none>
kube-proxy-kwj6m                                    1/1     Running             0          21m   172.17.0.4   pmem-csi-pmem-govm-worker2   <none>           <none>
kube-proxy-ndgg9                                    1/1     Running             0          21m   172.17.0.5   pmem-csi-pmem-govm-worker3   <none>           <none>
kube-proxy-s7tlk                                    1/1     Running             0          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
kube-scheduler-pmem-csi-pmem-govm-master            1/1     Running             2          25m   172.17.0.2   pmem-csi-pmem-govm-master    <none>           <none>
```

Reply:
I don't follow. Callico doesn't work when the endpoint feature is disabled? Is that always the case or just on 1.20? 

It makes sense to keep it disabled on older Kubernetes (because of the bug) and run with the defaults on Kubernetes 1.20 (because the bug is fixed); I'm just trying to understand whether that is the only reason for this change or there are other reasons.

Reply:
> I don't follow. Callico doesn't work when the endpoint feature is disabled? Is that always the case or just on 1.20?

It should be calico(or in general CNI plugin) issue, somehow `kube-proxy` it self trying access endpoints even if the feature is disabled, I need to check what was causing this.

Reply:
Is `kube-proxy` correctly configured to not use the feature when that happens?